idea/src/org/jetbrains/kotlin/idea/inspections/JavaCollectionsStaticMethodCallInspection.kt

```diff
@@ -0,0 +1,113 @@
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jetbrains.kotlin.idea.inspections
+
+import com.intellij.codeInspection.LocalQuickFix
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElementVisitor
+import org.jetbrains.kotlin.idea.caches.resolve.analyze
+import org.jetbrains.kotlin.idea.imports.importableFqName
+import org.jetbrains.kotlin.idea.intentions.callExpression
+import org.jetbrains.kotlin.js.descriptorUtils.nameIfStandardType
+import org.jetbrains.kotlin.load.java.descriptors.JavaMethodDescriptor
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.calls.callUtil.getResolvedCall
+import org.jetbrains.kotlin.resolve.calls.callUtil.getType
+import org.jetbrains.kotlin.resolve.lazy.BodyResolveMode
+
+class JavaCollectionsStaticMethodInspection : AbstractKotlinInspection() {
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+        return object : KtVisitorVoid() {
+            override fun visitDotQualifiedExpression(expression: KtDotQualifiedExpression) {
+                super.visitDotQualifiedExpression(expression)
+
+                val callExpression = expression.callExpression ?: return
+                val context = callExpression.analyze(BodyResolveMode.PARTIAL)
+                val resolvedCall = callExpression.getResolvedCall(context) ?: return
+                val descriptor = resolvedCall.resultingDescriptor as? JavaMethodDescriptor ?: return
+                val fqName = descriptor.importableFqName?.asString() ?: return
+                if (!canReplaceWithStdLib(fqName, callExpression, context)) return
+
+                holder.registerProblem(expression,
+                                       "Java Collections static method call can be replaced with Kotlin stdlib",
+                                       ProblemHighlightType.GENERIC_ERROR_OR_WARNING,
+                                       ReplaceWithSortFix(fqName))
+            }
+        }
+    }
+
+    private fun canReplaceWithStdLib(fqName: String, callExpression: KtCallExpression, context: BindingContext): Boolean {
+        if (!fqName.startsWith("java.util.Collections.")) return false
+        val valueArgs = callExpression.valueArguments
+        val valueArgsSize = valueArgs.size
+        if (valueArgsSize < 1) return false
+        if (!isMutableList(valueArgs[0], context)) return false
+        return when (fqName) {
```

Comment:
The `fill()` and `shuffle()` methods have been added in Kotlin 1.2. You need to check the API level for the current module, and offer these replacements only if the API level is at least 1.2.